src/components/flame-graph/FlameGraph.js

```diff
@@ -84,6 +84,61 @@ class FlameGraph extends React.PureComponent<Props> {
     }
   };
 
+  _handleKeyDown = (event: SyntheticKeyboardEvent<HTMLElement>) => {
+    const {
+      threadIndex,
+      flameGraphTiming,
+      callTree,
+      callNodeInfo: { callNodeTable },
+      selectedCallNodeIndex,
+      changeSelectedCallNode,
+    } = this.props;
+
+    if (selectedCallNodeIndex === null) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'ArrowDown': {
+        const prefix = callNodeTable.prefix[selectedCallNodeIndex];
+        if (prefix !== -1) {
+          changeSelectedCallNode(
+            threadIndex,
+            getCallNodePathFromIndex(prefix, callNodeTable)
+          );
+        }
+        break;
+      }
+      case 'ArrowUp': {
+        const [callNodeIndex] = callTree.getChildren(selectedCallNodeIndex);
+        if (callNodeIndex !== undefined) {
+          changeSelectedCallNode(
+            threadIndex,
+            getCallNodePathFromIndex(callNodeIndex, callNodeTable)
+          );
+        }
+        break;
+      }
+      case 'ArrowLeft':
+      case 'ArrowRight': {
+        const direction = event.key === 'ArrowLeft' ? -1 : 1;
+        const depth = callNodeTable.depth[selectedCallNodeIndex];
+        const row = flameGraphTiming[depth];
+        const rowIndex = row.callNode.indexOf(selectedCallNodeIndex);
+        const callNodeIndex = row.callNode[rowIndex + direction];
+        if (callNodeIndex !== undefined) {
+          changeSelectedCallNode(
+            threadIndex,
+            getCallNodePathFromIndex(callNodeIndex, callNodeTable)
+          );
+        }
```

Comment:
I don't really like that this code is using different data structures in order to navigate in different directions. `callNodeTable` up, `callTree` down, and `flameGraphTiming` and `callNodeTable` sideways. I initially tried to encapsulate this in `flamegraph.js`, but only working on `flameGraphTiming` would be inefficient since it would have to do a lot of search.
So I left it like this...

Reply:
I think it's good in the component actually :)